android/lib/src/main/java/com/cube/geofencing/RNRegionMonitorModule.java

```diff
@@ -0,0 +1,156 @@
+package com.cube.geofencing;
+
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import com.cube.geofencing.model.MonitoredRegion;
+import com.cube.geofencing.model.PersistableData;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableMap;
+import com.google.android.gms.common.api.ResultCallbacks;
+import com.google.android.gms.common.api.Status;
+import com.google.android.gms.location.Geofence;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+/**
+ * Defines the interface available from Javascript in order to manage monitored regions
+ */
+public class RNRegionMonitorModule extends ReactContextBaseJavaModule
+{
+	public static final String TAG = "RNRM";
+	public static final String TRANSITION_TASK_NAME = "region-monitor-transition";
+	public static final String REGION_SYNC_TASK_NAME = "region-monitor-sync";
+
+	private PersistableData data;
+	private GeofenceManager geofenceManager;
+
+	public RNRegionMonitorModule(@NonNull ReactApplicationContext reactContext)
+	{
+		super(reactContext);
+		data = PersistableData.load(reactContext);
+		geofenceManager = new GeofenceManager(reactContext);
+	}
+
+	@ReactMethod
+	public void addCircularRegion(@NonNull ReadableMap location, int radiusMetres, @NonNull String requestId, @NonNull final Promise promise)
+	{
+		try
+		{
+			Log.d(TAG, "addCircularRegion: " + requestId);
+
+			double latitude = location.getDouble("latitude");
+			double longitude = location.getDouble("longitude");
+			final MonitoredRegion region = new MonitoredRegion(requestId, latitude, longitude, radiusMetres);
+			Geofence geofence = region.createGeofence();
+			geofenceManager.addGeofences(Collections.singletonList(geofence), new ResultCallbacks<Status>()
+			{
+				@Override
+				public void onSuccess(@NonNull Status status)
+				{
+					Log.d(TAG, "addCircularRegion: " + status);
+
+					data.addRegion(region);
+					data.save(getReactApplicationContext());
+					promise.resolve(null);
```

Comment:
I believe an event is emitted, but not a callback from this function. It can take anywhere from a few seconds to a few minutes for the system to realise the user was already in the region.